### PR TITLE
feat(versioning/gradle): Support maven ranges in gradle files

### PR DESCRIPTION
**Pull request by Kouzukii**
## Changes

Maven version ranges like `[1.2.3,2)` are supported by gradle. 
Renovate handles these correctly for maven projects, for gradle projects however it does not create an update.
With this PR gradle version ranges will get updated just like maven's do.

## Context

We heavily use maven version ranges in our gradle projects and were surprised these were not updated by renovate.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/versioning/gradle/index.ts [hunk 1]
```diff
@@ -217,7 +218,7 @@ function getNewValue({
     }
   }
 
-  return null;
+  return mavenVersion.getNewValue({ currentValue, rangeStrategy, newVersion });
```
Comment (rarkins) [inline lib/modules/versioning/gradle/index.ts:221]: What happens here if the currentValue or newVersion aren't valid maven syntax - can that happen?

Comment (Kouzukii) [reply lib/modules/versioning/gradle/index.ts:221]: gradle supports all valid maven versions, any invalid maven version should simply return null

File: lib/modules/versioning/gradle/index.spec.ts
Comment (rarkins) [inline lib/modules/versioning/gradle/index.spec.ts:1]: Is it possible to get a test case which still returns null? or is that hard to manufacture?

Comment (Kouzukii) [reply lib/modules/versioning/gradle/index.spec.ts:1]: the maven versioning module does not return nulls for invalid versions, it just returns the `currentVersion`, but the resulting behavior should be identical to null
https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/process/lookup/index.ts#L367 

I could add some extra tests that verify that invalid versions result in `currentVersion` being returned

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
Thanks @Kouzukii for your PR!

**Pull request merged**